RATIONALE.md

```diff
@@ -0,0 +1,49 @@
+# Rationale for GitOps
+
+Currently, many software system's desired state is not defined separately from the running system.
+When we desire the behaviour of a software system to change, we modify the system's configuration directly, either through human action, or by running scripts that take a set of predetermined action on the system.
+
+This leads to several serious issues:
+
+- **Detecting drift from desired state**
+
+    If the desired state of a system is not explicitly defined, it is impossible to verify if the system in a correct state. The state of a running system itself does not provide sufficent information to determine its correctness. 
+    
+    Consider logging into an administration console and seeing that 28 machines are healthily running.
+    Is this good? Is this bad? That very much depends on what the desired number of machines is.
+    For example, these could be test machines that should have been deleted and are now incurring a significant cost for no reasons, or all that remains of a 100 machine datacenter.
+    We could consult the documentation, or expect the human operator to know, but by the time this occurs, our system has been in an incorrect state for a significant period of time. _The validity of the state of our system is not automatically verifiable_.
+
+    This problem leads to the conclusion that the canonical source of truth for desired state cannot be the actual state. 
+    Making the desired state of a system explicit prevents this problem.
+    
+- **Recovering from transitional states**
+
+    If we have no record of the desired state of our system, how can we recover from failures that occur in the transition between states? 
+    Such transitions are very common lifecycle events, such as upgrades, new features being released or scaling our resources. These are where the majority of hard software defects and transient errors occur.
+    Not only are such failures extremely common, but their likelihood grows algebraically with the number of components in our system.
```

Comment:
```suggestion
    Not only are such failures extremely common, but their likelihood grows exponentially with the number of components in our system.
```

Reply:
There's a very specific reason why that term is used, but the context for it is poor. I'll reword that paragraph to be clearer.

Reply:
@murillodigital does this paragraph reword make more sense now?